### PR TITLE
bpo-24234: Implement bytes.__bytes__

### PR DESCRIPTION
**Pull request by corona10**
https://bugs.python.org/issue24234

### CODE REVIEW

File: Tools/clinic/clinic.py [hunk 1]
```diff
@@ -2179,7 +2179,6 @@ def __repr__(self):
 __abs__
 __add__
 __and__
-__bytes__
```
Comment (corona10) [inline Tools/clinic/clinic.py:2182]: https://docs.python.org/3/c-api/typeobj.html?highlight=nb_int#sub-slots

Same reason as `__complex__`
https://github.com/python/cpython/pull/27886

**Review by gvanrossum — APPROVED** [review]
Should we perhaps combine the two News items for this bpo issue?

**Comment by corona10** [comment]
> Should we perhaps combine the two News items for this bpo issue?

Yes, it looks reasonable to add them as News items :) cc @mdickinson

**Comment by corona10** [comment]
````
doctest (doctest) ... 66 tests with zero failures
**********************************************************************
File "/home/runner/work/cpython/cpython/Lib/test/test_doctest.py", line 671, in test.test_doctest.test_DocTestFinder.non_Python_modules
Failed example:
    816 < len(tests) < 836 # approximate number of objects with docstrings
Expected:
    True
Got:
    False
**********

````

The broken test will be fixed after https://github.com/python/cpython/pull/27887 is merged and rebased.

**Comment by mdickinson** [comment]
> Should we perhaps combine the two News items for this bpo issue?

SGTM

**Review by mdickinson — APPROVED** [review]
LGTM, modulo the combining news items and the test failure.

File: Doc/whatsnew/3.11.rst [hunk 2]
```diff
@@ -169,6 +169,14 @@ Other Language Changes
   (Contributed by Serhiy Storchaka in :issue:`12022`.)
 
 
+Other CPython Implementation Changes
```
Comment (corona10) [inline Doc/whatsnew/3.11.rst:172]: @mdickinson Can you please take a look? Sorry, I am not a good English writer.

Comment (mdickinson) [reply Doc/whatsnew/3.11.rst:172]: @corona10 What you have looks fine to me!

**Comment by mdickinson** [comment]
Hmm. Unfortunately, I think those buildbot failures aren't false positives. By using `PyBytes_FromString`, we're truncating at the first zero byte. I think we need to use `PyBytes_FromStringAndSize` instead (and add a regression test).

```
>>> class B(bytes): pass
... 
>>> x = B(b"\x00\x01\x02")
>>> x.__bytes__()
b''
```

**Pull request merged**